src/codec.rs

```diff
@@ -1612,6 +1627,28 @@ mod tests {
 		assert_eq!((x, y), Decode::decode(&mut &encoded[..]).unwrap());
 	}
 
+	#[test]
+	fn boxed_str_works() {
+		let s = "Hello world".to_owned();
+		let b = s.clone().into_boxed_str();
+
+		let encoded = b.encode();
+		assert_eq!(s.encode(), encoded);
+
+		assert_eq!(*b, *Box::<str>::decode(&mut &encoded[..]).unwrap());
+	}
+
+	#[test]
+	fn boxed_slice_works() {
+		let v = vec![1u32, 2, 3, 4, 5, 6];
+		let b = v.clone().into_boxed_slice();
+
+		let encoded = b.encode();
+		assert_eq!(v.encode(), encoded);
+
+		assert_eq!(*b, *Box::<[u32]>::decode(&mut &b.encode()[..]).unwrap());
+	}
+
```

Comment:
You've added a blanket impl for `Box<[T]>`, so please also add a test for a more complicated variable length type `T`, e.g. maybe `Box<[String]>` or something like that.

Reply:
done